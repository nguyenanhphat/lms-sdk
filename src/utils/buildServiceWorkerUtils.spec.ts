describe('Exported functions', () => {
  test('Exported "appendImportScripts" function', () => {
    const { appendImportScripts } = require('./buildServiceWorkerUtils');
    expect(appendImportScripts).toBeInstanceOf(Function);
  });
  test('Exported "getDefinedInfoContent" function', () => {
    const { getDefinedInfoContent } = require('./buildServiceWorkerUtils');
    expect(getDefinedInfoContent).toBeInstanceOf(Function);
  });
  test('Exported "appendSourceMap" function', () => {
    const { appendSourceMap } = require('./buildServiceWorkerUtils');
    expect(appendSourceMap).toBeInstanceOf(Function);
  });
});

describe('Transform defined to code', () => {
  test('Success cases', () => {
    const { getDefinedInfoContent } = require('./buildServiceWorkerUtils');
    const codeTransformed = getDefinedInfoContent({
      version: '"__version__"',
      platform: '"__platform__"',
    });
    expect(codeTransformed).toMatchSnapshot();
  });

  describe('Failed cases', () => {
    const SAMPLES = [1, null, undefined, new Map(), new Set(), () => {}];

    const { default: ensureSlash } = require('./ensureSlash');
    test.each(SAMPLES)('Pass %j should throw error', (path: any) => {
      function run() {
        ensureSlash(path, true);
      }
      expect(run).toThrowErrorMatchingSnapshot();
    });
  });
});
