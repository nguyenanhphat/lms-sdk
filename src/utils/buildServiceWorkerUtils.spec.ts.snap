// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform defined to code Failed cases Pass {} should throw error 1`] = `"Path received type is object not is a string"`;

exports[`Transform defined to code Failed cases Pass {} should throw error 2`] = `"Path received type is object not is a string"`;

exports[`Transform defined to code Failed cases Pass 1 should throw error 1`] = `"Path received type is number not is a string"`;

exports[`Transform defined to code Failed cases Pass null should throw error 1`] = `"Path received type is object not is a string"`;

exports[`Transform defined to code Failed cases Pass undefined should throw error 1`] = `"Path received type is undefined not is a string"`;

exports[`Transform defined to code Failed cases Pass undefined should throw error 2`] = `"Path received type is function not is a string"`;

exports[`Transform defined to code Success cases 1`] = `"self.version = \\"__version__\\";self.platform = \\"__platform__\\";"`;
